### PR TITLE
[refactor] multiprocess_pipe: remove pipelined_backward

### PR DESCRIPTION
**Pull request by msbaines**
We want to limit the number of __init__ params. If pipelined_backward
is important, we'll add it back but not make it an option.

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by min-xu-ai — APPROVED** [review]
it seems that this is a workaround anyway. to be clear, you still plan to have 3D parallelism like:

DDP (Pipeline(tensor-parallel()))

right?

**Comment by msbaines** [comment]
> it seems that this is a workaround anyway. to be clear, you still plan to have 3D parallelism like:
> 
> DDP (Pipeline(tensor-parallel()))
> 
> right?

Looks like its needed. Updating to use this feature for MulitProcessPipe but not for AsyncPipe.

**Comment by sidgoyal78** [comment]
> Looks like its needed. Updating to use this feature for MulitProcessPipe but not for AsyncPipe.


Yes, this is needed for multi process.

File: tests/nn/pipe_process/test_pipe.py [hunk 1]
```diff
@@ -765,24 +753,6 @@ def __init__(self, module):
...
-def pipelined_backward(pipe_class):
-    model = nn.Sequential(nn.ReLU(), nn.ReLU())
-
-    destroy_model_parallel()
-    initialize_model_parallel(1, 4)
-    pipe = pipe_class(model, [1, 1], worker_map=get_worker_map())
-
-    assert pipe.pipelined_backward is False
-
-    destroy_model_parallel()
```
Comment (sidgoyal78) [inline tests/nn/pipe_process/test_pipe.py:779]: this part could be useful for MultiProcessPipe?

**Pull request merged**